### PR TITLE
New permission for create project

### PR DESCRIPTION
**Pull request by aeng**
https://bugzilla.redhat.com/show_bug.cgi?id=1214502

### CODE REVIEW

File: zanata-war/src/main/java/org/zanata/security/SecurityFunctions.java [hunk 1]
```diff
@@ -88,7 +88,7 @@ public static boolean canManageRoles(String target) {
     /* Any authenticated user can create a project */
```
Comment (davidmason) [inline zanata-war/src/main/java/org/zanata/security/SecurityFunctions.java]: Update comment to match the new behaviour.

File: zanata-war/src/main/java/org/zanata/util/EssentialDataCreator.java [hunk 2]
```diff
@@ -106,6 +106,14 @@ public void prepare() {
                 adminExists = false;
             }
 
+            if (!accountRoleDAO.roleExists("project-creator")) {
+                log.info("Creating 'project-creator' role");
+                if (accountRoleDAO.create("project-creator", MANUAL, "user") == null) {
+                    throw new RuntimeException(
+                        "Couldn't create 'project-creator' role");
```
Comment (davidmason) [inline zanata-war/src/main/java/org/zanata/util/EssentialDataCreator.java]: "Could not" (avoid contractions in technical writing).

**Comment by davidmason** [comment]
:+1:

File: zanata-war/src/main/java/org/zanata/util/EssentialDataCreator.java [hunk 3]
```diff
@@ -101,11 +101,19 @@ public void prepare() {
...
             }
 
+            if (!accountRoleDAO.roleExists("project-creator")) {
+                log.info("Creating 'project-creator' role");
+                if (accountRoleDAO.create("project-creator", MANUAL, "user") == null) {
+                    throw new RuntimeException(
+                        "Could not create 'project-creator' role");
+                }
+            }
+
```
Comment (seanf) [inline zanata-war/src/main/java/org/zanata/util/EssentialDataCreator.java]: I think we are at (or past) the point where this bit of logic should be extracted into a method, called once for each role we want.  (Or a for loop.)

**Comment by djansen-redhat** [comment]
:white_check_mark: Tested

**Comment by aeng** [comment]
Merged into integration/master. https://github.com/zanata/zanata-server/commit/7e90a9b74ecb888710e67bbd0a87d4058e00699a

**Comment by seanf** [comment]
Shouldn't the squash merge 7e90a9b have a name like "feat(role): New permission for create project" instead of "feat: role: New permission for create project"?

**Comment by aeng** [comment]
@seanf you're right.

**Pull request closed without merging**